server/container/util/src/test/java/org/apache/james/util/StreamUtilsTest.java

```diff
@@ -112,4 +113,18 @@ void ofNullableShouldReturnAStreamWithElementsOfTheArray() {
             .collect(Guavate.toImmutableList()))
             .containsExactly(1, 2);
     }
+
+    @Test
+    void unfoldShouldGenerateAnInfiniteStream() {
```

Comment:
what happens when function always return Optional and what happens when there's actually an infinite stream?